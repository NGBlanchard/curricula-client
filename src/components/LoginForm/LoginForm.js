import React, { Component } from 'react'
import CourseApiService from '../../services/course-api-service';
import TokenService from '../../services/token-service';

import { Button, Input } from '../Utils/Utils'

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {
    }
  }

  state = { error: null }

  handleAuth = e => {
    e.preventDefault()
    this.setState({ error: null })
    const { user_name, password } = e.target
    CourseApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })
    .then(res => {
      user_name.value = ''
      password.value = ''
      TokenService.saveAuthToken(res.authToken)
      this.props.onLoginSuccess()
    })
    .catch(res => {
      this.setState({ error: res.error })
    })
  }


  render() {
    const { error } = this.state
    return (
      <form
        className='LoginForm'
        onSubmit={this.handleAuth}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='user_name'>
          <label htmlFor='LoginForm__user_name'>
            User name
          </label>
          <Input
            name='user_name'
            id='LoginForm__user_name'>
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='LoginForm__password'>
            Password
          </label>
          <Input
            name='password'
            type='password'
            id='LoginForm__password'>
          </Input>
        </div>
        <Button type='submit'>
          Login
        </Button>
      </form>
    )
  }
}
