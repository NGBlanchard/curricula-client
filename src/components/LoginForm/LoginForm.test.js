import LoginForm from './LoginForm'
import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });

describe('<LoginForm />', () => {
  it('Renders without crashing', () => {
    mount(<LoginForm/>)
  })
})
