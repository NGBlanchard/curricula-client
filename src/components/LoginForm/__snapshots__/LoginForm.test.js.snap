// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders Login Form by default 1`] = `
<LoginForm
  onLoginSuccess={[Function]}
>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="user_name"
    >
      <label
        htmlFor="LoginForm__user_name"
      >
        Username
      </label>
      <Input
        id="LoginForm__user_name"
        name="user_name"
      >
        <input
          className="Input "
          id="LoginForm__user_name"
          name="user_name"
        />
      </Input>
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <Input
        id="LoginForm__password"
        name="password"
        type="password"
      >
        <input
          className="Input "
          id="LoginForm__password"
          name="password"
          type="password"
        />
      </Input>
    </div>
    <Button
      type="submit"
    >
      <button
        className="Button "
        type="submit"
      >
        Login
      </button>
    </Button>
  </form>
</LoginForm>
`;
